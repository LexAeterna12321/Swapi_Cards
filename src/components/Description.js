import React from "react";
import PropTypes from "prop-types";
import { CardDescription } from "./styledComponents";

const Description = ({ name, birth_year, gender }) => (
  <CardDescription>
    <h3>Name: {name}</h3>
    <p>Birth Date: {birth_year}</p>
    <p>Gender: {gender}</p>
  </CardDescription>
);

Description.propTypes = {
  name: PropTypes.string.isRequired,
  birth_year: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired
};

export default Description;
