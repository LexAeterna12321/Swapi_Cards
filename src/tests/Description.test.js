import React from "react";
import Description from "../components/Description";
import { shallow } from "enzyme";
it("expects to render SwapiCharCard component", () => {
  expect(shallow(<Description />)).toMatchSnapshot();
});
