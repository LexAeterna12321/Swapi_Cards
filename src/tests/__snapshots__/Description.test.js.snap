// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render SwapiCharCard component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Name: 
        </h3>,
        <p>
          Birth Date: 
        </p>,
        <p>
          Gender: 
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Birth Date: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Birth Date: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Gender: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Gender: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "rules": Array [
          "
  padding: 20px;
  text-transform: uppercase;
",
        ],
      },
      "displayName": "styled.section",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "section",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Name: 
          </h3>,
          <p>
            Birth Date: 
          </p>,
          <p>
            Gender: 
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Birth Date: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Birth Date: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Gender: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Gender: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "rules": Array [
            "
  padding: 20px;
  text-transform: uppercase;
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
